src/pad_tail.rs

```diff
@@ -13,14 +12,16 @@ pub struct PadUsing<I, F> {
     iter: Fuse<I>,
     min: usize,
     pos: usize,
+    back: usize,
+    total_len: usize,
```

Comment:
But wouldn't this suffice:

```
struct PadUsing {
 iter,
 filler,
 elements_from_next: usize, // no. of elements generated by calls to `next` that's what's currently pos
 elements_from_next_back: usize, // no. of elements generated by calls to `next_back`
 elements_required: usize, // no. of elements required by pad_using caller
}
```

In my naive view, this should be sufficient to determine at each step what to generate next. (And it avoids the `min`/`total_len` dichotomy).

Reply:
applied this change